File: modules/wem-api/src/main/java/com/enonic/wem/api/content/page/region/PartComponentType.java
package com.enonic.wem.api.content.page.region;

public final class PartComponentType
    extends ComponentType
{
    public final static PartComponentType INSTANCE = new PartComponentType();

    private static final PartComponentDataSerializer dataSerializer = new PartComponentDataSerializer();

    private PartComponentType()
    {
        super( "part", PartComponent.class );
    }

    @Override
    public PartComponentDataSerializer getDataSerializer()
    {
        return dataSerializer;
    }
}

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/content/page/region/PartComponentType.ts
module api.content.page.region {

    export class PartComponentType extends ComponentType {

        private static INSTANCE: PartComponentType = new PartComponentType();

        constructor() {
            super("part");
        }

        newComponentBuilder(): PartComponentBuilder {
            return new PartComponentBuilder();
        }

        public static get(): PartComponentType {
            return PartComponentType.INSTANCE;
        }
    }

}
